### PR TITLE
[SPARK-48820][SQL][DOC] Correct the examples for Collate functions

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
This PR proposes to correct the examples for `Collate` functions.

### Why are the changes needed?
According to the code in `FunctionRegistry` show below.
```
    expressionBuilder("collate", CollateExpressionBuilder),
    expression[Collation]("collation"),
```

, `SELECT COLLATION('Spark SQL' collate UTF8_LCASE);` and `CollateExpressionBuilder` are not related.
In fact, `SELECT COLLATION('Spark SQL' collate UTF8_LCASE);` is related to `Collation`.

### Does this PR introduce _any_ user-facing change?
'No'.
New feature.

### How was this patch tested?
GA.

### Was this patch authored or co-authored using generative AI tooling?
'No'.

### CODE REVIEW

**Comment by beliefer** [comment]
ping @dbatomic @mihailom-db @stefankandic

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collationExpressions.scala [hunk 1]
```diff
@@ -38,7 +38,7 @@ import org.apache.spark.sql.types._
     Examples:
       > SET spark.sql.collation.enabled=true;
       spark.sql.collation.enabled	true
-      > SELECT COLLATION('Spark SQL' _FUNC_ UTF8_LCASE);
+      > SELECT _FUNC_('Spark SQL', 'UTF8_LCASE');
       UTF8_LCASE
```
Comment (stefankandic) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collationExpressions.scala:42]: the output of the `collate` function will be the first argument (`Spark SQL`) and not the given collation

Comment (beliefer) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collationExpressions.scala]: Thank you.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collationExpressions.scala [hunk 2]
```diff
@@ -38,8 +38,8 @@ import org.apache.spark.sql.types._
     Examples:
       > SET spark.sql.collation.enabled=true;
       spark.sql.collation.enabled	true
-      > SELECT COLLATION('Spark SQL' _FUNC_ UTF8_LCASE);
-      UTF8_LCASE
+      > SELECT _FUNC_('Spark SQL', 'UTF8_LCASE');
+      spark sql
```
Comment (stefankandic) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collationExpressions.scala:42]:
```suggestion
      Spark SQL
```
Collating a string to lowercase collation does not mean to convert it to lowercase characters. 

Lowercase collation just means that strings will be compared as equal if their lowercase variants are the same, but the string representation does not change.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collationExpressions.scala [hunk 3]
```diff
@@ -38,8 +38,8 @@ import org.apache.spark.sql.types._
     Examples:
       > SET spark.sql.collation.enabled=true;
       spark.sql.collation.enabled	true
-      > SELECT COLLATION('Spark SQL' _FUNC_ UTF8_LCASE);
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collationExpressions.scala:41]: This example is fine. We need a way to show the specified string collation.

Comment (cloud-fan) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collationExpressions.scala:41]: And `str COLLATE UTF8_LCASE` is the more standard way to specify collation.

Comment (beliefer) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collationExpressions.scala:41]: In fact, this example is not related to `CollateExpressionBuilder`.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collationExpressions.scala (hunk 2 above)
Comment (beliefer) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collationExpressions.scala:42]: Got it.

**Pull request closed without merging**